R/data.R

```diff
@@ -29,7 +29,7 @@ S7_data <- function(object) {
 `S7_data<-` <- function(object, check = TRUE, value) {
   attrs <- attributes(object)
   object <- value
-  attributes(object) <- attrs
+  attributes(object) <- modify_list(attrs, attributes(value))
```

Comment:
Maybe it should only modify attributes that aren't props? I'm not sure that matters in practice, but seems like the right principle?

Reply:
As I think about this, I see it's trickier than I first thought. What if a user declares a base attribute as a property? For example:
```r
MyList := new_class(class_list, properties = list(
  names = class_character
))
MyArray := new_class(class_double, properties = list(
  dim = class_integer
))
```
Right now, `S7_data()` drops `names` and `dim`, and `` `S7_data<-`() `` doesn't update them, neither of which is what a user would likely want.
Maybe we should encourage more explicit usage of `unclass()` and `` `attributes<-`() `` to "temporarily" retrieve and update the objects?

Reply:
Or maybe using `name` or `dim` as a property should be an error?

Reply:
Also, `S7_data()` completely removes the `class` attribute, which breaks some base S3 classes like factor, POSIXct, etc.

Reply:
I think that's to be expected? The goal of `S7_data` is to give you the underlying base object. OTOH if you're extending an S3 class, it seems like there will be case where you want to get the underlying S3 object.

Reply:
Oh, I might be misunderstanding the role of `S7_data()` then. I was thinking of it as and 'un_S7()` and `re_S7()<-`.

Reply:
I only reached for it because `?super` suggested it as the way to do `NextMethod()` in an S3 generic method. Since `S7_data()` drops the class attribute, we may want to update `?super` docs.